runtime/drivers/duckdb/model_manager.go

```diff
@@ -0,0 +1,176 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelInputProperties struct {
+	SQL string `mapstructure:"sql"`
+}
+
+func (p *ModelInputProperties) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	return nil
+}
+
+type ModelOutputProperties struct {
+	Table               string                      `mapstructure:"table"`
+	Materialize         *bool                       `mapstructure:"materialize"`
+	UniqueKey           []string                    `mapstructure:"unique_key"`
+	IncrementalStrategy drivers.IncrementalStrategy `mapstructure:"incremental_strategy"`
+}
+
+func (p *ModelOutputProperties) Validate(opts *drivers.ModelExecutorOptions) error {
+	if opts.Incremental {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("incremental models must be materialized")
+		}
+		p.Materialize = boolPtr(true)
+	}
+
+	if opts.InputConnector != opts.OutputConnector {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("models that output to a different connector must be materialized")
+		}
+		p.Materialize = boolPtr(true)
+	}
+
+	switch p.IncrementalStrategy {
+	case drivers.IncrementalStrategyUnspecified, drivers.IncrementalStrategyAppend, drivers.IncrementalStrategyMerge:
+	default:
+		return fmt.Errorf("invalid incremental strategy %q", p.IncrementalStrategy)
```

Comment:
**Issue**: The error message for an invalid incremental strategy does not provide the valid options, which could lead to confusion. <br> **Fix**: Update the error message to include the valid incremental strategies to guide the user in correcting their input. <br> **Code Suggestion**: 
 ```
 - return fmt.Errorf("invalid incremental strategy %q", p.IncrementalStrategy)
 + return fmt.Errorf("Invalid incremental strategy %q. Valid strategies are 'Unspecified', 'Append', and 'Merge'.", p.IncrementalStrategy)
 ```